app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,609 @@
+                } else {
+                    multiselectSelectedItems[itemKey] = item as DialogOption;
+                }
+                setMultiselectSelected(multiselectSelectedItems);
+            }
+        }
+    }, [integrationData, multiselectSelected, isMultiselect, dataSource, handleSelect]);
+
+    const handleRemoveOption = (item: UserProfile | Channel | DialogOption) => {
+        const currentSelected: Dictionary<UserProfile> | Dictionary<DialogOption> | Dictionary<Channel> = multiselectSelected;
+        const itemKey = extractItemKey(dataSource, item);
+        const multiselectSelectedItems = {...currentSelected};
+        delete multiselectSelectedItems[itemKey];
+        setMultiselectSelected(multiselectSelectedItems);
+    };
+
+    const getChannels = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {channels: channelData} = await fetchChannels(serverUrl, currentTeamId, page.current);
+
+            setLoading(false);
+
+            if (channelData && channelData.length > 0) {
+                setIntegrationData(channelData);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const getProfiles = useCallback(debounce(async () => {
+        if (next.current && !loading && !term) {
+            setLoading(true);
+            page.current += 1;
+
+            const {users: profiles} = await fetchProfiles(serverUrl, page.current);
+
+            setLoading(false);
+
+            if (profiles && profiles.length > 0) {
+                setIntegrationData(profiles);
+            } else {
+                next.current = false;
+            }
+        }
+    }, 100), [integrationData, loading, term]);
+
+    const loadMore = useCallback(async () => {
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            await getProfiles();
+        } else if (dataSource === ViewConstants.DATA_SOURCE_CHANNELS) {
+            await getChannels();
+        }
+
+        // dynamic options are not paged so are not reloaded on scroll
+    }, [getProfiles, getChannels, dataSource]);
+
+    const searchDynamicOptions = useCallback(async (searchTerm = '') => {
+        if (options) {
+            setIntegrationData(options);
+        }
+
+        if (!getDynamicOptions) {
+            return;
+        }
+
+        const results: DialogOption[] = await getDynamicOptions(searchTerm.toLowerCase());
+        const searchData = results || [];
+
+        if (searchTerm) {
+            setSearchResults(searchData);
+        } else {
+            setIntegrationData(searchData);
+        }
+    }, [options, getDynamicOptions]);
+
+    const onHandleMultiselectSubmit = useCallback(() => {
+        handleSelect(getMultiselectData());
+        close();
+    }, [multiselectSelected, handleSelect]);
```

Comment:
Why this depends on `multiselectSelected`?

Reply:
refactored so it's easier to see the dep